### ISSUE TITLE
fails to compile on macos

### ISSUE THREAD

**Issue by ZtereoHYPE** [MrGlockenspiel/activate-linux#9]
activate macos functiontnality please i need it fro my macos mac

**Comment by StereoMadnessss** [comment]
This is for linux, as such this does not work on MacOS. As I said, this is made for linux and not MacOS.

**Comment by ZtereoHYPE** [comment]
:( 0 star s bad sfotware

**Comment by MrGlockenspiel** [comment]
I would port it but i havent used a mac in years, much less owned one, and I know nothing about development on them, nor do I particularly want to

**Comment by MrGlockenspiel** [comment]
If someone does manage to make it cross platform feel free to submit a PR

**Comment by Snivyer1910** [comment]
I'mma make it but for Windows

**Comment by Arkhtek** [comment]
Noooo dont port it to macos let them suffer

**Comment by MrGlockenspiel** [comment]
Probably compiles on MacOS now, but wont run because I need to strip out the code that makes it require 32 bit color support

**Comment by Devnol** [comment]
Can confirm, compiles and segfaults macOS

**Comment by Devnol** [comment]
Not really good at debugging but I tried running it inside lldb and the output was as follows:
```
(lldb) target create "bin/activate_macos"
run
Current executable set to '/Users/pakou/Documents/CS_Projects/activate-linux/bin/activate_macos' (x86_64).
(lldb) run
Process 84575 launched: '/Users/pakou/Documents/CS_Projects/activate-linux/bin/activate_macos' (x86_64)
Process 84575 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0xe8)
    frame #0: 0x0000000100003a5a activate_macos`main + 42
activate_macos`main:
->  0x100003a5a <+42>: movq   0xe8(%rcx), %rax
    0x100003a61 <+49>: movslq 0xe0(%rcx), %rcx
    0x100003a68 <+56>: shlq   $0x7, %rcx
    0x100003a6c <+60>: movq   0x10(%rax,%rcx), %rax
Target 0: (activate_macos) stopped.
```

**Comment by MrGlockenspiel** [comment]
I put a workaround for the 32 bit color support a couple days ago, might make a VM and fix it on there sometime

**Comment by Ruby-Dragon** [comment]
not sure if installing from makefile is supported on Mac. I have never owned a mac so I really don't know, but that may be something to work on.

**Comment by Devnol** [comment]
Make install is basically just copying files to places so yeah it is supported as long as you have the right perms. Not sure how that would affect it though, since the output binary isn't installed anywhere

**Comment by Ruby-Dragon** [comment]
I know some people with macs and might be able to get them to test out this

**Comment by IAMACAR10** [comment]
i have a friend with a mac and i wanna look into testing this. also need to change the text to say 'activate macOS'

**Comment by Devnol** [comment]
It already does that, see [here](https://github.com/MrGlockenspiel/activate-linux/blob/827e5ee5f53c617155a9cc7059fc8f9e55b0cddd/src/activate_linux.c#L98)

**Comment by ZtereoHYPE** [comment]
I have already tested and everything works if not for transparency for lack of support of it in macOS's XQuartz and the fact that it does not stay on top of all windows for non standard implementation of X11.

**Comment by ZtereoHYPE** [comment]
This issue is left open only so people see that it doesn't work perfectly.

**Comment by smthlow** [comment]
I installed xorg-server, cairo and xorg-libXinerama with ports, restarted shell, changed flags to `CFLAGS  = -L/opt/local/lib -lX11 -lXfixes -lXinerama -lcairo -I/opt/local/include/cairo -I/opt/local/include`, it compiles, but crashes with segfault when I run it.

I tried to debug it with gdb but it it says `??` instead of file names, so idk where it crashes.

If you can guide me how to debug it better, I can help. I haven't worked with c.

**Comment by MrGlockenspiel** [comment]
Yeah these are all because of the non-standard X implementation on macOS and there's nothing I can really do

**Comment by Fedeparg** [comment]
It runs for me, bot does not show text. Compiled with the flags above. Did I do anything wrong?
<img width="491" alt="image" src="https://user-images.githubusercontent.com/28566295/168618192-83db60c8-688d-4b8e-90f4-d94b24e20b89.png">

**Comment by MikeHingley22** [comment]
> Yeah these are all because of the non-standard X implementation on macOS and there's nothing I can really do

could you copy the window pixels for that areas, and then use that as a bitmap to render the text onto (not an expert - just wondering if it is something possible)

**Comment by Ruby-Dragon** [comment]
That plus a gausian blur is how windows does acrylic

**Comment by Naville** [comment]
> It already does that, see [here](https://github.com/MrGlockenspiel/activate-linux/blob/827e5ee5f53c617155a9cc7059fc8f9e55b0cddd/src/activate_linux.c#L98)
@Devnol FYI it should be "System Preferences"
<img width="190" alt="image" src="https://user-images.githubusercontent.com/5205699/168732896-8a5775b8-047a-4330-a22c-e9c7c673ffec.png">

**Comment by b3z** [comment]
Pls I need this so I can finally throw away my windows bootcamp partition.

**Comment by Naville** [comment]
[Lakr233/ActivateMac](https://github.com/Lakr233/ActivateMac) FYI

**Comment by Mega-Touk** [comment]
Pls port it to Chromebook now

**Comment by Ruby-Dragon** [comment]
Chromebooks are linux. Just turn on linux mode and install

**Comment by IAMACAR10** [comment]
yeah, chrome OS is pretty much just gentoo behind the scenes, so anything
that works on gentoo (i believe) should work on chrome os.

(also i didn't leave e-mail reply in this time 😎)

**Comment by rollingmoai** [comment]
If this is going to be cross-platform, the repo should be renamed to something like `activate-watermark`. Windows support is now available so we are nearly there.

**Comment by Ruby-Dragon** [comment]
i mean, i guess, but Linux is still the best supported platform for this project

**Comment by MrGlockenspiel** [comment]
everything outside of linux is an afterthought, plus activate-linux just sounds better and is funnier

**Comment by MrGlockenspiel** [comment]
I don't have a mac to test it on but I suspect a sticky note would work

**Comment by Arkhtek** [comment]
activate-macos fork when

**Comment by MrGlockenspiel** [comment]
yup sure was
